OpenSim/Common/Component.h

```diff
@@ -198,19 +198,16 @@ OpenSim_DECLARE_ABSTRACT_OBJECT(Component, Object);
     Component(const std::string& aFileName,
         bool aUpdateFromXMLNode = true) SWIG_DECLARE_EXCEPTION;
 
+    /** Construct Component as a subcomponent member of another Component.
+        Automatically marks the component as a subcomponent of the owner.
+        A subcomponent MUST be named. **/
+    explicit Component(const std::string& name, Component* owner=nullptr);
```

Comment:
I think I agree with @sherm1's comments, but I also think I know why @aseth1 made this constructor. The constructor is supposed to be used during the in-class member initialization of a member subcomponent. Since the in-class member initializer is called during construction of the parent (owner), is it permissible to call a method on the parent as a side effect of the in-class member initialization?